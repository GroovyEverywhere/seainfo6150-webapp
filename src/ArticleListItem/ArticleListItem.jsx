import React from "react";
import HTMLText from "../HTMLText/HTMLText";

const ArticleListItem = (props) => {
  return (
    <header>
      {/* this is an example of how you insert props into a react component */}
      <h1>{props.article.title}</h1>
      
      <time datetime = {props.article.timeStamp}> {props.article.displayDate}</time>

      <HTMLText text={props.article.shortText} />
    </header>
  );
};
ArticleListItem.propTypes = {
};

export default ArticleListItem;
