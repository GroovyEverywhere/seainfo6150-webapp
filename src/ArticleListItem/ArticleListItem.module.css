.container {
    background-image: linear-gradient(to right, rgba(204, 0, 0, .5), #eee 20% ) ;
    border: 1px solid #c00;
    border-radius: 10px;
    box-shadow: 0 5px 10px 5px rgba(0, 0, 0, .3);
    font-family: Arial, Helvetica, sans-serif;
    padding: 10px;
    
}

.container {
    margin-top: 30px;
    margin-left: 20px;
}

.title {
    text-decoration: underline;
    color: #c00;
    margin: 0 0 20px;
}

.text {
    font-size: 12px;
    margin: 0 0 20px;
}

.time {
    font-weight: bold;
}

@media screen and (max-width: 1200px) {
    .article{
        display: flex;
        flex-direction: column;
        justify-content: space-between;
        height: 100%;
    }
    .container{
        flex-basis: 25%;
    }
}

@media screen and (max-width: 900px) {
    .article{
        display: flex;
        flex-direction: column;
        justify-content: space-between;
        height: 100%;
    }
    .container{
        flex-basis: 40%;
    }
}