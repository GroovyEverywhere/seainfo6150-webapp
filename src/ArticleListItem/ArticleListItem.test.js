import React from "react";
import ArticleListItem from "./ArticleListItem.jsx";
import { fireEvent } from "@testing-library/react";

describe("ArticleListItem tests", () => {
  it("renders correctly when text is showing", () => {
    const { container } = render(<li />);
    expect(container).toMatchSnapshot();
  });

});
