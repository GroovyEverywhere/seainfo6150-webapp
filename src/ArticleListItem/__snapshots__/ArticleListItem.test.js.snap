// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleListItem tests renders correctly when text is showing 1`] = `
<div>
  <li />
</div>
`;
